templates/common/_base/files/mtu-migration-dispatcher.yaml

```diff
@@ -0,0 +1,213 @@
+        local dev=$1
+        mtu_path="/sys/class/net/${dev}/mtu"
+        if [ ! -f "$mtu_path" ]; then
+          echo "ERROR: no mtu file found at $mtu_path" >&2
+          return 1
+        fi
+        cat $mtu_path
+    }
+
+    set_mtu_on_dev(){
+        local dev=$1
+        local c_type=$2
+        local new_mtu=$3
+        if [[ $c_type =~ .*ovs.* ]]; then
+            ovs-vsctl set int $dev mtu_request=$new_mtu
+        else
+            ip link set dev $dev mtu $new_mtu
+        fi
+    }
+
+    set_mtu_on_dev_routes() {
+        local dev=$1
+        local mtu=$2
+
+        dev_routes=$(ip route show dev $dev)
+
+        if [ -n "$dev_routes" ]; then
+            while IFS= read -r line; do
+                current_mtu=$(echo $line | sed -n -e 's/^.*mtu \([^ ]*\).*/\1/p')
+                echo "mtu=$current_mtu in this route: $line"
+                if [ -z "$current_mtu" ] || [ $current_mtu -gt $mtu ]; then
+                    ip route change ${line} dev $dev mtu ${mtu}
+                fi
+            done <<< "$dev_routes"
+        fi
+    }
+
+    save_mtu_for_dev(){
+        local dev=$1
+        local mtu=$2
+
+        if [ -f "${MTU_MIGRATION_SAVE_DIR}/${dev}" ]; then
+            rm -f "${MTU_MIGRATION_SAVE_DIR}/${dev}"
+        fi
+        if [ ! -d "${MTU_MIGRATION_SAVE_DIR}" ]; then
+            mkdir "${MTU_MIGRATION_SAVE_DIR}"
+        fi
+        echo "ORIGINAL_MTU=$mtu" > "${MTU_MIGRATION_SAVE_DIR}/${dev}"
+    }
+
+    set_mtu(){
+        local dev=$1
+        local c_type=$2
+        local is_vlan_parent=$3
+
+        if [ -f "${MTU_MIGRATION_SAVE_DIR}/${dev}" ]; then
+            . "${MTU_MIGRATION_SAVE_DIR}/${dev}"
+        else
+            ORIGINAL_MTU=$(get_mtu_on_dev "$dev")
+            save_mtu_for_dev "$dev" "$ORIGINAL_MTU"
+        fi
+
+        echo "For ORIGINAL_MTU=$ORIGINAL_MTU and TARGET_MTU=$TARGET_MTU ..."
+        echo "with is_vlan_parent=$is_vlan_parent"
+        if [ $ORIGINAL_MTU -gt $TARGET_MTU -a $is_vlan_parent -eq 0 ]; then
+            routable_mtu=$TARGET_MTU
+            mtu=$ORIGINAL_MTU
+        elif [ $ORIGINAL_MTU -lt $TARGET_MTU ]; then
+            routable_mtu=$ORIGINAL_MTU
+            mtu=$TARGET_MTU
+            echo "set routable_mtu=$routable_mtu and mtu=$mtu"
+        elif [ $is_vlan_parent -eq 1 ]; then
+            echo "No need to set MTU as current is greater than target and is a vlan parent"
+            return 0
+        else
+             echo "No need to set MTU as current and target are equal"
+             return 0
+        fi
+
+        echo "set routable_mtu=$routable_mtu and mtu=$mtu"
+        set_mtu_on_dev "$dev" "$c_type" "$mtu"
+        set_mtu_on_dev_routes "$dev" "$routable_mtu"
+    }
+
+    iface="$1"
+    echo "running on an UP event for host interface $iface"
+
+    if ! [ -f "$MTU_MIGRATION_CONFIG" ]; then
+        echo "no ongoing MTU migration. Nothing to do."
+        set +x
+        exit
+    fi
+
+    if [ -z "$CONNECTION_UUID" ]; then
```

Comment:
where does CONNECTION_UUID come from?

Reply:
It's among the environment variables set by network manager when calling the dispatcher script. I will add a comment that clarifies that and the input arguments as well.